lib/services.js

```diff
@@ -117,15 +123,48 @@ async function startService (udid, serviceName, socket, handshakeOnly = false) {
     } else {
       return await connectPort(udid, service.Port, socket);
     }
+  } catch (e) {
+    if (!tryMountIfFailed || serviceName === MOBILE_IMAGE_MOUNTER_SERVICE_NAME) {
+      throw e;
+    }
+    if (e.message?.includes('InvalidService') && await autoMountDeveloperImage(udid)) {
+      return startService(udid, serviceName, socket, handshakeOnly, false);
+    }
   } finally {
     lockdown.close();
   }
 }
 
+/**
+ * mount developer image based on device version and local storage.
+ * @param {string} udid
+ * @returns If mount success
```

Comment:
```suggestion
 * @returns {Promise<boolean>} If mount success
```

but why resolving a `boolean` instead of just rejecting?

Reply:
I thought type of return is auto-recognized so I don't have to doc it. That's maybe a bad habit
And return a boolean can make code easy to use like 
```nodejs
if (e.message?.includes('InvalidService') && await autoMountDeveloperImage(udid)) {
      return startService(udid, serviceName, socket, handshakeOnly, false);
    }
```
Since this is inside a catch block of another part.

But maybe this should be recoded to check mount status before start service if tryMountIfFailed is `true` and make value of tryMountIfFailed from `opts` from user.
The auto fetch part should also get image from github-repo/url which is assigned by user. 
I should fix them later. I'm quite busy at work these days. I hope such days won't last too long 😔.